### DISCUSSION TITLE
Dynamic routes in nextjs got null

### DISCUSSION THREAD

**Discussion by devmnj** [keystonejs/keystone#6582] [Event Q&A Q3 2021]
I am trying to get dynamic route  in  nextjs-keystone5 app. And got 
TypeError: Cannot read property 'query' of null

import { useRouter } from 'next/router'

const Post = () => {
  const router = useRouter()
  const { pid } = router.query

  return <p>Post: {pid}</p>
}

export default Post

**Comment by bladey** [comment]
Hey @manojap,

Thanks for your question! It looks like you're stuck making progress with your project.

To get the most out of the live event, we're keeping this Community Q&A category focused on questions that are relevant to the community as a whole.

Seeing as this question is specific to your project, would you mind posting it to our community Slack in the #keystone-5 channel?

That's the best place to get focused support on blocks you encounter with Keystone - https://community.keystonejs.com
